userland/libtock/ble.h

```diff
@@ -2,125 +2,120 @@
+#define BLE_CFG_TX_POWER_CMD      2
+#define BLE_CFG_ADV_ITV_CMD       3
+#define BLE_ADV_CLEAR_DATA_CMD    4
+#define BLE_SCAN_CMD              5
+#define BLE_SCAN_SUBSCRIBE        0
+#define BLE_CFG_ADV_ADDR_ALLOW    0x30
+#define BLE_CFG_SCAN_BUF_ALLOW    0x31
 
 
-// TX power
-// FIXME: Not platform independent.
 typedef enum {
-    POS4_DBM     = 0x04,
-    ODBM         = 0x00,
-    NEG4_DBM     = 0xFC,
-    NEG_8_DBM    = 0xF8,
-    NEG_12_DBM   = 0xF4,
-    NEG_16_DBM   = 0xF0,
-    NEG_20_DBM   = 0xEC,
-    NEG_30_DBM   = 0xD8
-} BLE_TX_Power_t;
+  GAP_FLAGS                              = 0x01, /* Flags, see enum below */
+  GAP_INCOMPLETE_LIST_16BIT_SERVICE_IDS  = 0x02, /* Incomplete list of 16-bit Service IDs. */
+  GAP_COMPLETE_LIST_16BIT_SERVICE_IDS    = 0x03, /* Complete list of 16-bit Service IDs. */
+  GAP_INCOMPLETE_LIST_32BIT_SERVICE_IDS  = 0x04, /* Incomplete list of 32-bit Service IDs (not relevant for Bluetooth 4.0). */
+  GAP_COMPLETE_LIST_32BIT_SERVICE_IDS    = 0x05, /* Complete list of 32-bit Service IDs (not relevant for Bluetooth 4.0). */
+  GAP_INCOMPLETE_LIST_128BIT_SERVICE_IDS = 0x06, /* Incomplete list of 128-bit Service IDs. */
+  GAP_COMPLETE_LIST_128BIT_SERVICE_IDS   = 0x07, /* Complete list of 128-bit Service IDs. */
+  GAP_SHORTENED_LOCAL_NAME               = 0x08, /* Shortened Local Name. */
+  GAP_COMPLETE_LOCAL_NAME                = 0x09, /* Complete Local Name. */
+  GAP_TX_POWER_LEVEL                     = 0x0A, /* TX Power Level (in dBm). */
+  GAP_DEVICE_ID                          = 0x10, /* Device ID. */
+  GAP_SLAVE_CONNECTION_INTERVAL_RANGE    = 0x12, /* Slave Connection Interval Range. */
+  GAP_LIST_128BIT_SOLICITATION_IDS       = 0x15, /* List of 128 bit service UUIDs the device is looking for. */
+  GAP_SERVICE_DATA                       = 0x16, /* Service Data. */
+  GAP_APPEARANCE                         = 0x19, /* Appearance */
+  GAP_ADVERTISING_INTERVAL               = 0x1A, /* Advertising Interval. */
+  GAP_MANUFACTURER_SPECIFIC_DATA         = 0xFF  /* Manufacturer Specific Data. */
+} GapAdvertisementData_t;
+
+enum {
+  LE_LIMITED_DISCOVERABLE = 0x01, /* Peripheral device is discoverable for a limited period of time. */
+  LE_GENERAL_DISCOVERABLE = 0x02, /* Peripheral device is discoverable at any moment. */
+  BREDR_NOT_SUPPORTED     = 0x04, /* Peripheral device is LE only. */
+  SIMULTANEOUS_LE_BREDR_C = 0x08, /* Not relevant - central mode only. */
+  SIMULTANEOUS_LE_BREDR_H = 0x10 /*Not relevant - central mode only. */
+} GapFlags_t;
+
+
+
+
+/*******************************************************************************
+ *   User-space API
+ ******************************************************************************/
+
+
+ // initialize advertisement (should be used before invoken ble_start_advertising)
+ //
+ // advertising_itv_ms  - advertisement interval in milliseconds
+ // flags               - flags in the advertisement defined in GapFlags_t enum
```

Comment:
Why expose flags to the user rather than just selecting the right ones? For instance, users will _always_ have to specify `BREDR_NOT_SUPPORTED`.

I would suggest replacing this with a `is_connectable` bool and then have the code automatically select the flags.

Reply:
Sure it is a better idea